### PR TITLE
Printout mapping fixes

### PR DESCRIPTION
**Pull request by david-shortman**
Completes #24 

Updates class roster and sign in sheets to be 
- generated by class id
- inclusive of more student data mappings
- filtered for summer 2023

### CODE REVIEW

File: libs/firebase/functions/src/lib/utilities/functions.utility.ts [hunk 1]
```diff
@@ -49,10 +50,11 @@ class FunctionBuilder<SecretNames extends string, StringNames extends string> {
         return new FunctionBuilder(this.secrets, this.strings, roles);
     }
 
-    handle<RequestData>(
+    handle<RequestData, QueryData extends ParsedQs = ParsedQs>(
         handler: (
-            request: Omit<functions.https.Request, 'body'> & {
+            request: Omit<functions.https.Request, 'body' | 'query'> & {
                 body: { data: RequestData };
+                query: QueryData;
```
Comment (david-shortman) [inline libs/firebase/functions/src/lib/utilities/functions.utility.ts:57]: just updated TS stuff here

File: libs/firebase/student/reports/src/lib/utilities/roster-report.utilities.ts [hunk 2]
```diff
@@ -126,73 +126,80 @@ export class RosterReportGenerator {
...
-                        ?.join(', '),
-                    metadata: {
-                        isImportant: false,
+            .map((student) => {
+                console.log('student', student.guardians);
+                return {
+                    lastName: { value: student.last_name },
+                    firstName: { value: student.first_name },
+                    age: {
+                        value: (() => {
```
Comment (david-shortman) [inline libs/firebase/student/reports/src/lib/utilities/roster-report.utilities.ts:135]: IIFE! (https://developer.mozilla.org/en-US/docs/Glossary/IIFE)

File: libs/firebase/student/reports/src/lib/utilities/roster-report.utilities.ts [hunk 3]
```diff
@@ -147,7 +147,10 @@ export class RosterReportGenerator {
                     })(),
                 },
                 guardianContacts: {
-                    value: '',
+                    value:
+                        student.guardians
+                            ?.map(this.contactToString)
+                            .join('\n') ?? '',
```
Comment (david-shortman) [inline libs/firebase/student/reports/src/lib/utilities/roster-report.utilities.ts:153]: Guardians are now included in roster

**Review by katiez07 — APPROVED** [review]
verrrra nice :D

**Pull request merged**